playbooks/ha_log_cfg_sync.yaml

```diff
@@ -0,0 +1,138 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+---
+- name: Config rsync daemon on log server slave node
+  hosts: logserver-slave, zuul-scheduler-slave
+  become: yes
+  tasks:
+    - file: path=/etc/rsyncd state=directory
+
+    - file: path=/var/run/rsyncd.pid state=absent
+
+    - name: Create rsyncd config file for logs and zuul config files sync
+      template:
+        src: "{{ labkeeper_config_git_dest }}/rsync/rsyncd.conf.j2"
+        dest: /etc/rsyncd/rsyncd.conf
+
+    - name: Start rsync daemon on zuul slave node
+      shell: |
+        echo 'root:openlab' > /etc/rsyncd/rsyncd.pass
+        chmod 600 /etc/rsyncd/rsyncd.pass
+        rsync --daemon --config=/etc/rsyncd/rsyncd.conf
+      args:
+        executable: /bin/bash
+
+- name: Config rsync client on zuul master node
+  hosts: logserver-master, zuul-scheduler-master
+  become: yes
+  tasks:
+    - file: path=/etc/rsync state=directory
+
+    - name: Create sync password file
+      copy:
+        dest: /etc/rsync/rsync_client.pwd
+        content: "openlab"
+        mode: 0600
+
+    - name: Create script for sync log files
+      copy:
+        dest: /etc/rsync/cron_sync_zuul_log.sh
```

Comment:
Sorry, I am a little confused, we need to create sync log script on zuul and log server nodes? I think we only have to create the scripts and add crontab tasks on zuul master node? Or I missed something?

Reply:
when developing the playbooks, I recommend that we'd better not only consider the current deployment situation. we should make the playbook more extension possibilities. that is why you can see there are groups: `zuul-scheduler`, `zuul-web`, `zuul-executor`... and match to the plays. we can simply define the group for all the zuul service to a `zuul` group, because currently we deployed all the services on 1 host. but it is flexible, someday if we want to deploy only `zuul-scheduler` in a separated host, we only need to modify the inventory file based on current playbooks implement.